### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Vogel612**
Small fixes for Command Calling, Additional Command information, Reply functionality, minor refactorings

### CODE REVIEW

File: src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java [hunk 1]
```diff
@@ -58,7 +61,10 @@ private void processMessage(final ChatMessage chatMessage) {
         final String trigger = AppContext.INSTANCE.get(BotConfig.class).getTrigger();
         if ( !chatMessage.getMessage().startsWith(trigger)) { return; }
 
-        commands.stream().filter(c -> chatMessage.getMessage().replace(trigger, "").startsWith(c.getName())).findFirst().map(c -> c.execute(chatMessage))
+        final String commandName = chatMessage.getMessage().replace(trigger, "").split(" ")[0];
```
Comment (Vincentyification) [inline src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java]: Wait, just discovered a problem. Considering that we're using `.replace()` method which replaces all occurrences of a literal sequence, wouldn't this tear away parts of the message if it includes the trigger as well? `!!h!!i` -> `hi` - I think we need `replaceFirst`.

Comment (Vogel612) [inline src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java]: Holy carp. You're fully correct! Lucky we have a review ;)

Comment (mangstadt) [inline src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java]: Or better yet:

    chatMessage
    .getMessage()
    .replaceAll("^" + Pattern.quote(trigger), "")
    .split(" ")[0]

Comment (Vincentyification) [inline src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java]: @mangstadt Yeah but you're still using `.replaceAll`, it's that bit simpler to use `.replaceFirst`, as it tells the regex engine to just replace first and not to keep trying (which always fails for the second attempt).

Comment (Vogel612) [inline src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java]: That in turn would make it quite moot to use an anchor and the code then would be:

    chatMessage.getMessage().replaceFirst(Pattern.quote(trigger), "").split(" ")[0];

File: src/main/java/com/gmail/inverseconduit/bot/Program.java [hunk 2]
```diff
@@ -51,66 +43,28 @@
...
         LOGGER.info("Startup completed.");
     }
 
-    private void bindDefaultCommands() {
-        bindNumberCommand();
-        bindJavaDocCommand();
-    }
-
-    private void bindNumberCommand() {
-        final Pattern p = Pattern.compile("^\\d+$");
```
Comment (Vincentyification) [inline src/main/java/com/gmail/inverseconduit/bot/Program.java]: Considering how we're using this pattern for anchored matching, we could make it `"\\d+"` and replace the line down there with `matcher.matches()`.

**Comment by Vogel612** [comment]
http://chat.stackoverflow.com/transcript/message/21299359#21299359

**Pull request merged**